### PR TITLE
Add material on relaxed init and pointers in proof

### PR DESCRIPTION
**Pull request by yannickmoy**

### CODE REVIEW

File: courses/spark_for_ada_programmers/110_advanced_proof_techniques.rst [hunk 1]
```diff
@@ -498,6 +506,181 @@ Does your loop terminate?
...
+
+-------------------------
+Borrowing and Observing
+-------------------------
+
+* Borrowing is temporary read-write access
+
+  - either through a declaration
+
+    X : access Cell := Current_Cell.Next;
```
Comment (leogermond) [inline courses/spark_for_ada_programmers/110_advanced_proof_techniques.rst:628]: Missing the `.. code:: Ada` section

File: courses/spark_for_ada_programmers/110_advanced_proof_techniques.rst [hunk 2]
```diff
@@ -498,6 +506,181 @@ Does your loop terminate?
...
+
+  - or through call
+
+    procedure Update_Cell (X : access Cell);
```
Comment (leogermond) [inline courses/spark_for_ada_programmers/110_advanced_proof_techniques.rst:632]: ditto

File: courses/spark_for_ada_programmers/110_advanced_proof_techniques.rst [hunk 3]
```diff
@@ -498,6 +506,181 @@ Does your loop terminate?
...
+    Update_Cell (Current_Cell.Next);
+
+* Observing is temporary read-only access
+
+  - either through a declaration
+
+    X : access constant Cell := Current_Cell.Next;
```
Comment (leogermond) [inline courses/spark_for_ada_programmers/110_advanced_proof_techniques.rst:639]: Ditto

File: courses/spark_for_ada_programmers/110_advanced_proof_techniques.rst [hunk 4]
```diff
@@ -498,6 +506,181 @@ Does your loop terminate?
...
+
+  - or through call
+
+    procedure Read_Cell (X : access constant Cell);
```
Comment (leogermond) [inline courses/spark_for_ada_programmers/110_advanced_proof_techniques.rst:643]: Ditto

**Review by leogermond — CHANGES_REQUESTED** [review]
Missing some code blocks

File: courses/spark_for_ada_programmers/110_advanced_proof_techniques.rst (hunk 2 above)
Comment (yannickmoy) [reply courses/spark_for_ada_programmers/110_advanced_proof_techniques.rst]: fixed

File: courses/spark_for_ada_programmers/110_advanced_proof_techniques.rst (hunk 3 above)
Comment (yannickmoy) [reply courses/spark_for_ada_programmers/110_advanced_proof_techniques.rst]: fixed

File: courses/spark_for_ada_programmers/110_advanced_proof_techniques.rst (hunk 4 above)
Comment (yannickmoy) [reply courses/spark_for_ada_programmers/110_advanced_proof_techniques.rst]: fixed

**Review by leogermond — APPROVED** [review]

**Pull request merged**